src/structures/GuildChannel.js

```diff
@@ -156,6 +156,14 @@ class GuildChannel extends Channel {
     return [...sorted.values()].indexOf(sorted.get(this.id));
   }
 
+  /**
+   * Checks if the channel is not viewable by users with only the @everyone role.
+   * @type {boolean} true if private, false otherwise.
+   */
+  get private() {
+    return !this.permissionsFor(this.guild.roles.everyone).has(Permissions.FLAGS.VIEW_CHANNEL);
```

Comment:
```suggestion
    return !this.permissionsFor(this.guild.id).has(Permissions.FLAGS.VIEW_CHANNEL);
```

Reply:
I'd argue that this suggestion's code feels hacky, and does the same steps.

- If `Snowflake` -> Get `?Role` -> `permissionsFor(Role?)` 
- Get `Role` -> If `Role` -> `permissionsFor(Role)`

Only difference is that `permissionsFor(Snowflake)`'s return type is nullable (because the case of invalid role IDs) whereas `permissionsFor(Role)` is not, so in the event when we rewrite to TypeScript, this would be one nullish assertion operator less.

Also, in the event Discord ever changes the `@everyone`'s role ID to be something else, `roles.everyone` would be updated and this wouldn't need further changes, but `this.guildId` would.